Expand ecobee documentation for set_hold_mode

dbrowndan opened the pull request:
Add some notes explaining how to access your custom ecobee Comfort Settings when using the 'set_hold_mode' service
Source: My own experimentation

**Description:**
Just minor documentation tweaks.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#<home-assistant PR number goes here>

## Checklist:

- [x] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

frenck reviewed (dismissed):
Since this PR changes the currently existing documentation, the target branch should be `current`.
Please change the target and rebase if needed.

Commit:
Expand ecobee documentation for set_hold_mode service

Added some notes explaining how to access your custom ecobee "Comfort Settings" when using set_hold_mode

dbrowndan commented:
Thanks for bearing with me. This is my first ever attempt to commit to a major open-source project. Learning to rebase now. Closed by accident, reopening.

frenck commented:
Well... you did a fine job @dbrowndan 👍 
Marking PR ready for review.

DubhAd commented on source/_components/climate.ecobee.markdown:
```diff
@@ -245,7 +245,7 @@ canceled.
 | ---------------------- | -------- | ----------- |
 | `entity_id` | yes | String or list of strings that point at `entity_id`'s of climate devices to control. Else targets all.
-| `hold_mode` | no | 'temp', 'home', 'away', 'sleep', etc., None
+| `hold_mode` | no | `temp`, `home`, `away`, `sleep`, `None`, etc. NOTE: If you create custom hold modes (also known as "Comfort Settings") on your ecobee.com dashboard, their hold_modes are 'smart1', 'smart2', 'smart3', etc. The number for each custom mode should match the mode's icon on your ecobee.com dashboard. Also note that the mode numbers/icons in the ecobee mobile app MAY NOT MATCH the n ...
```
This makes the table hard to read IMO - it'd be better having that as a follow on paragraph after the table.

dbrowndan committed:
Move ecobee hold_mode notes below table

Per https://github.com/home-assistant/home-assistant.io/pull/5980#discussion_r209421798

dbrowndan commented:
Thanks for looking things over, @DubhAd! Changes committed.

DubhAd commented on source/_components/climate.ecobee.markdown:
```diff
@@ -245,7 +245,9 @@ canceled.
-| `hold_mode` | no | 'temp', 'home', 'away', 'sleep', etc., None
+| `hold_mode` | no | `temp`, `home`, `away`, `sleep`, `None`, `smart1`, `smart2`, etc.
+
+NOTE: If you create custom hold modes (also known as "Comfort Settings") on your ecobee.com dashboard, their hold_modes are `smart1`, `smart2`, `smart3`, etc. The number for each custom mode should match the mode's icon on your ecobee.com dashboard. Also note that the mode numbers/icons in the ecobee mobile app MAY NOT MATCH the numbers/icons from the ecobee.com web dashboard. The ones on the web ...
```
Sorry, missed the SHOUTING - please use italics for emphasis

dbrowndan replied:
Sure thing, updated. Thanks again.

dbrowndan committed:
Use italics for emphasis

Per https://github.com/home-assistant/home-assistant.io/pull/5980#discussion_r210516058

DubhAd reviewed (approved):
👍
